src/os/unix/x11.rs

```diff
@@ -115,10 +115,30 @@ impl DisplayInfo {
                 return Err(Error::WindowCreate("XOpenDisplay failed".to_owned()));
             }
 
-            let screen = (lib.XDefaultScreen)(display);
-            let visual = (lib.XDefaultVisual)(display, screen);
+            let screen;
+            let visual;
+            let depth;
+
+            let mut vinfo: xlib::XVisualInfo = std::mem::zeroed();
+            //FIXME: X_PutImage
+            if transparency {
+                (lib.XMatchVisualInfo)(
+                    display,
+                    (lib.XDefaultScreen)(display),
+                    32,
+                    xlib::TrueColor,
+                    &mut vinfo as *mut _,
+                );
+                depth = vinfo.depth;
+                visual = vinfo.visual;
+                screen = vinfo.screen;
```

Comment:
Just a style nit, but would be nice to follow declaration order above and assignment order below:
```suggestion
                screen = vinfo.screen;
                visual = vinfo.visual;
                depth = vinfo.depth;
```